tests/standard/test_file_source.py

```diff
@@ -0,0 +1,62 @@
+import dlt
+import pytest
+
+from sources.standard.filesystem import filesystem_resource
+from tests.utils import ALL_DESTINATIONS, assert_load_info, load_table_counts
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_all_resources(destination_name: str) -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="file_source",
+        destination=destination_name,
+        dataset_name="file_source_data",
+        full_refresh=True,
+    )
+
+    # Load all files
+    all_files = filesystem_resource(filename_filter="*.csv")
+    load_info = pipeline.run(all_files)
+    assert_load_info(load_info)
+    table_counts = load_table_counts(pipeline, "filesystem")
+    assert table_counts["filesystem"] == 4
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_filtered_resources(destination_name: str) -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="file_source",
+        destination=destination_name,
+        dataset_name="file_source_data",
+        full_refresh=True,
+    )
+
+    # Load filter files
+    all_files = filesystem_resource(filename_filter="mlb*.csv")
+    load_info = pipeline.run(all_files)
+    assert_load_info(load_info)
+    table_counts = load_table_counts(pipeline, "filesystem")
+    assert table_counts["filesystem"] == 2
```

Comment:
the filesystem_resource should be deselected by default and it should not load the file items (IMO) - but we need to say in docs how to enable it

please check the csv table as well